### DISCUSSION TITLE
question on the decomposition calculation

### DISCUSSION THREAD

**Discussion by julialiu7** [google/meridian#286] [General]
How is the MMM decomposition calculated given the posterior distributions of the coefficients and transformation parameters? I can imagine it involves not only the distribution aspect of the model parameters (betas, alphas, and ec), the dependent variable  has also been scaled by population and standardized. Given those complexities, could you please outline some kind of step by step calculation. That will be greatly appreciated. Thanks.

**Comment by brendalewisprice** [comment]
Clearly you have thought carefully about how Meridian standardizes and transforms inputs prior to fitting the model. This is to reduce the varied influence from inputs that are on a variety of different scales. Post model fit, outputs are transformed back to the original scale. To put it another way, the model, using the transformed data, estimates the transformed incremental impact of each media channel and then transforms it back to it's original scale.

In our [documentation](https://developers.google.com/meridian/docs/basics/input-data) you can see what terms are transformed and how. The expression used to to signify the transformation is "L" as described in this [page](https://developers.google.com/meridian/docs/basics/input-data#transformation_kpi_units). For an example of how the transformation back to the original scale is done, see the documentation on the [fixed budget optimization](https://developers.google.com/meridian/docs/basics/optimization-without-reach-frequency#fixed_budget_optimization).

**Reply by julialiu7** [reply]
Thank you for the answer. Good to know that things are 'reversed back' at the end. 
Could you please comment/answer the other part in my original question? That is how is, for example, decay used exactly in the transformation data. The decay posterior is a distribution, how is it used in transformation? Do you use the mean/median or the whole distribution to transform the data? if the whole distribution, how exactly is it done?

**Comment by jyklly** [comment]
The MMM decomposition is based on the incremental impact of each media channel and is calculated using the [_compute_pct_of_contribution](https://github.com/google/meridian/blob/6efbaeb543f543effd5ce4f02b71e78b8c6f768d/meridian/analysis/analyzer.py#L3906) function. This function takes in an input [incremental_impact_posterior](https://github.com/google/meridian/blob/6efbaeb543f543effd5ce4f02b71e78b8c6f768d/meridian/analysis/analyzer.py#L3909C7-L3909C35) which is an aggregate incremental impact for each media channel. Note this is an aggregate impact and is derived from calculating the incremental impact for each posterior draw. This is done in the [incremental_impact](https://github.com/google/meridian/blob/6efbaeb543f543effd5ce4f02b71e78b8c6f768d/meridian/analysis/analyzer.py#L1086) function and via the helper function [_incremental_impact_impl](https://github.com/google/meridian/blob/6efbaeb543f543effd5ce4f02b71e78b8c6f768d/meridian/analysis/analyzer.py#L983). The transformation back to the original scale is handled by the [_inverse_impact](https://github.com/google/meridian/blob/6efbaeb543f543effd5ce4f02b71e78b8c6f768d/meridian/analysis/analyzer.py#L944) function which is called by [_incremental_impact_impl](https://github.com/google/meridian/blob/6efbaeb543f543effd5ce4f02b71e78b8c6f768d/meridian/analysis/analyzer.py#L983). So, in short, we first perform all reverse transformations (scale, adstock, hill) using the entire set of posterior parameter samples, and then take the mean of the posterior predicted values in order to calculate the percent contribution of each channel.

**Reply by vsamaga** [reply]
How do we display the control variable contribution as part of the plot_contribution_waterfall_chart()? I think it's naturally being aggregated under 'Baseline' as of now.

**Reply by gousesb** [reply]
Hi @vsamaga ,

Thank you for contacting us.

We would like to inform you that the `plot_contribution_waterfall_chart()` function does not currently support breaking down the control variable contribution. This is because, by design, the contribution of control variables is aggregated and displayed as part of the 'Baseline' in the waterfall chart.

Control variables are aggregated under the 'Baseline' because their purpose is to neutralize confounding factors, not to be a focus of the analysis. By combining their contribution into a single, foundational component, the chart remains focused on the primary variables of interest, while still accounting for the controlled factors' influence. If you wish to learn more about reasons controls don't have causal inference or baseline breakdown, check our documentation [here](https://developers.google.com/meridian/docs/advanced-modeling/control-variables#rsns).

Feel free to reach out if you have any questions or suggestions.

Google Meridian Support Team
